pagefaultgames/pokerogue
# [Localization] Use gender context for achievments/achv

## What are the changes the user will see?

Effectively none

## Why am I making these changes?

Right now we have the inconvenient situation of having a `achv-male.json` and `achv-female.json`.
I wanted to reduce the redundancy by merging those by using the [i18n `context`](https://www.i18next.com/translation-function/context) key.

## What are the changes from a developer perspective?

Since `en` is the single source of truth for the language keys all `achv.name` that have need a different key for female translations had to be added as an empty `"name_female": null` entry.

<sub>This change might also create new translation "voids" in [GitLocalize](https://gitlocalize.com/repo/9839). That's unfortunately a necessary evil</sub>

To be able to merge `male` and `female` achv files I used the file-compare tool of my IDE. All Asian languages had no differences between m & f.

### Screenshots/Videos
<sup>Don't worry about the `invalid date`. This is because I "cheater" the achievments</sup>
<img width="350" alt="image" src="https://github.com/user-attachments/assets/5300dbe4-b326-4d09-808e-c340d6074cec"><img width="350" alt="image" src="https://github.com/user-attachments/assets/8351ce2f-1a63-4eeb-8026-5caf1dd9b439">

## How to test the changes?
- Start the game in a language that has gender specific achievment names
    - e.g. `de`, `es`, `pt_BR`, `fr`
- Check the achievements (I hope you have some unlocked. I have no data with many unlocked :(... If you have some feel free to share)
- Change your gender to a different one
- Check them again => Should render the gender context string

## Checklist
